include/SFML/System/Angle.inl

```diff
@@ -0,0 +1,218 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2022 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+namespace priv
+{
+    constexpr float pi = 3.141592654f;
+
+    constexpr float fmodf(float x, float y)
+    {
+        while (x >= y)
+            x -= y;
+        return x;
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+constexpr Angle::Angle() :
+m_radians(0)
+{
+}
+
+
+////////////////////////////////////////////////////////////
+constexpr float Angle::asDegrees() const
+{
+    return m_radians * 180 / priv::pi;
+}
+
+
+////////////////////////////////////////////////////////////
+constexpr float Angle::asRadians() const
+{
+    return m_radians;
+}
+
+
+////////////////////////////////////////////////////////////
+constexpr Angle::Angle(float radians)
+    : m_radians(0.0f)
+{
+    // Normalize to [0, 2*pi)
+    while (radians < 0.0f)
+        radians += 2 * priv::pi;
```

Comment:
You're not wrong haha. Surely there's a better way to do this same calculation but I'm not sure what that would be.

Reply:
I mean, I just showed an implementation of `fmodf()` that works for negative numbers.

Reply:
Oh it does? How exactly would this loop get replaced with `fmodf()`? I can't picture how that would work. Something like this?
```cpp
constexpr Angle::Angle(float radians)
    : m_radians(0.0f)
{
    // Normalize to [0, 2*pi)
    m_radians = priv::fmodf(radians, 2 * priv::pi);
    if (radians < 0.0f)
        m_radians += 2 * priv::pi;
}
```

Reply:
Just this:
```cpp
    m_radians = priv::fmodf(radians, 2 * priv::pi);
```

By the way, I think this method should not be called `fmodf()` since it has different semantics and never returns negative numbers. `signedRemainer()` would  be more appropriate.

I used this snippet to run a few tests, it's probably a good idea if we add this to unit tests as well. Might even become part of our "tools" header.
```cpp
#include<iostream>
#include<cassert>

using namespace std;

float unsignedRemainder(float a, float b)
{
    assert(b > 0);
    if (a >= 0)
        return a - static_cast<int>(a / b) * b;
    else
        return a + b - static_cast<int>(a / b) * b;
}

int main()
{
    cout<< unsignedRemainder(90, 40) << endl;
    cout<< unsignedRemainder(-90, 40) << endl;
    cout<< unsignedRemainder(50, 40) << endl;
    cout<< unsignedRemainder(-50, 40) << endl;
    cout<< unsignedRemainder(30, 40) << endl;
    cout<< unsignedRemainder(-30, 40) << endl;
    cout<< unsignedRemainder(0, 40) << endl;
    
    // should assert:
    cout<< unsignedRemainder(90, -40) << endl;
    cout<< unsignedRemainder(-90, -40) << endl;
    
    return 0;
}
```

Reply:
I'm reluctant to test `priv::fmodf` (or whatever it's name should be since it's no longer analogous to `std::fmodf`) because it's an implementation detail and not part of the public API. That changes if it get elevated out of the `priv::` namespace but currently that's not the case.

Reply:
It works! Thanks for helping me get rid of all the loops.